test/unit/lib/auth-client.js

```diff
@@ -332,6 +365,58 @@ test('getAuthUrl - single select junk', function(t) {
   t.is(actual, expected);
 });
 
+test('getAuthUrl - single select with junk keys in object passed in', function(t) {
```

Comment:
Thoughts on how singleSelect should behave? (we should test for all of this porbably, I think we can do it in 2 tests only
```
singleSelect: true                    -> single_select=true
singleSelect: false/'hello'/0/1/3     -> single_select=false
singleSelect: undefined or not passed -> single_select is not added

singleSelect: {}, {potato}            -> single_select=false
singleSelect: {vin}, {vin, potato}    -> single_select=true&single_select_vin=vin
```

Reply:
this is the behavior that I have imagined! the only ones that aren't covered currently are `{}` and `{vin, potato}`. Can add these cases as well!